example/conv2DWithMdspan/src/conv2DWithMdspan.cpp

```diff
@@ -0,0 +1,243 @@
+    //! \param filter Filter-matrix
+
+    template<typename TAcc, typename MdSpan>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc, MdSpan const input, MdSpan output, MdSpan const filter) const
+        -> void
+    {
+        auto matrixWidth = input.extent(0);
+        auto matrixHeight = input.extent(1);
+        // Filter matrix is square
+        auto filterWidth = filter.extent(0);
+
+        // Get thread index, the center of filter-matrix is positioned to the item on this index.
+        auto const [row, col] = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+
+        // The convolutional filter-matrix applied to the input matrix, it's position is row and col. If some of the
+        // items of the filter are outside the matrix, those are not taken into calculation or they are assumed zero.
+        if(col < matrixWidth && row < matrixHeight)
+        {
+            float pValue{0.0f};
+            for(int32_t fRow = 0; fRow < static_cast<int32_t>(filterWidth); fRow++)
+            {
+                for(int32_t fCol = 0; fCol < static_cast<int32_t>(filterWidth); fCol++)
+                {
+                    // Position of input matrix element to be multiplied with the corresponding element at filter
+                    auto const exactRow = static_cast<int32_t>(row) - filterWidth / 2 + fRow;
+                    auto const exactCol = static_cast<int32_t>(col) - filterWidth / 2 + fCol;
+                    if(exactRow >= 0 && exactRow < matrixHeight && exactCol >= 0 && exactCol < matrixWidth)
+                    {
+                        pValue += filter(fRow, fCol) * input(exactRow, exactCol);
+                    }
+                }
+            }
+            output(row, col) = pValue;
+        }
+    }
+};
+
+auto FuzzyEqual(float a, float b) -> bool
+{
+    return std::fabs(a - b) < std::numeric_limits<float>::epsilon() * 1000.0f;
+}
+
+// In standard projects, you typically do not execute the code with any available accelerator.
+// Instead, a single accelerator is selected once from the active accelerators and the kernels are executed with the
+// selected accelerator only. If you use the example as the starting point for your project, you can rename the
+// example() function to main() and move the accelerator tag to the function body.
+template<typename TAccTag>
+auto example(TAccTag const&) -> int
+{
+    // Define the index domain
+    using Dim = alpaka::DimInt<2>;
+    // Index type
+    using Idx = std::uint32_t;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    // Define the accelerator
+    using DevAcc = alpaka::TagToAcc<TAccTag, Dim, Idx>;
+    using QueueAcc = alpaka::Queue<DevAcc, alpaka::NonBlocking>;
+
+    using DataType = float;
+    static constexpr Idx filterWidth = 5;
+    static constexpr Idx matrixWidth = 128;
+    static constexpr Idx matrixHeight = 128;
+
+    static_assert(
+        alpaka::Dim<DevAcc>::value == 2u,
+        "The accelerator used for the Alpaka Kernel has to be 2 dimensional!");
```

Comment:
This seems redundant here, given that the `DevAcc` has a dimension of 2 by construction.

Reply:
Though it could make sense to have this check inside the kernel.

Reply:
done, thanks.